File: src/main/java/ru/scarlet/filestorage/AttachmentType.java
package ru.scarlet.filestorage;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Getter
public enum AttachmentType {
    DOCUMENT(0, "DOCUMENT"),
    PERSONAL(1, "PERSONAL"),
    PUBLIC(2, "PUBLIC");

    private int code;
    private String attachmentType;
    AttachmentType(int code, String attachemntType) {
        this.code=code;
        this.attachmentType=attachemntType;
    }

    private AttachmentType getByCode(int code){
        if (code==DOCUMENT.getCode()) return AttachmentType.DOCUMENT;
        if (code== PERSONAL.getCode()) return AttachmentType.PERSONAL;
        if (code== PUBLIC.getCode()) return AttachmentType.PUBLIC;
        return null;
    }
}

File: src/main/java/ru/scarlet/filestorage/enums/AttachmentType.kt
package ru.scarlet.filestorage.enums

import lombok.Getter

@Getter
enum class AttachmentType(val code: Int, val attachmentType: String) {
    DOCUMENT(0, "DOCUMENT"),
    PERSONAL(1, "PERSONAL"),
    PUBLIC(2, "PUBLIC");

    private fun getByCode(code: Int): AttachmentType? {
        return when (code) {
            DOCUMENT.code -> DOCUMENT
            PERSONAL.code -> PERSONAL
            PUBLIC.code -> PUBLIC
            else -> null
        }
    }
}
